Allow users to name folders in the UI

JTSIV1 opened the pull request:
Fixes #

## Proposed Changes

  - [x] Adds UI elements to job submission and past results  pages to allow the user to set folder names
  - [x] Folder names are reflected and updated in the UI (past jobs page and your jobs pages)
  - [x] When folders are downloaded they include the user assigned name (as well as the previously included jobID information)
  - [x] Folders are displayed in reverse order so more recent jobs are at the top
  - [x] Add naming rules to [this](https://github.com/cybergis/cybergis-compute-python-sdk/blob/85a36626b24be85444e2073d80978ace3f1fd5be/cybergis_compute_client/UI.py#L544) location

JTSIV1 committed:
Added naming functionality

Added functionality to name jobs when submitting and when viewing the "Past Results" tab of the UI. When files are named and then downloaded, their name will be included in the download folder's name.

JTSIV1 committed:
Change from hard-coded put function to request in Client.py

This removes the hard-coded put function and replaces its uses with calls to client.request

JTSIV1 committed:
Made name filtering more robust and reversed folders displayed

-Folder names are now made to contain only safe characters through a separate function. All folders will now be downloaded with only safe character options.
-On the past job screen, the list of past jobs was reversed so that more recent jobs are on top.
-Removed extra imports

JTSIV1 committed:
Added pagination to Past Results page

Created pagination for the past results page, only displaying 10 jobs at a time and allowing the user to navigate multiple pages instead of a long list

mitkotak commented on cybergis_compute_client/UI.py:
```diff
@@ -817,6 +910,11 @@ def rerender(self, components=[]):
 
+    """ Used to ensure that folders have names with only safe characters """
+    def makeNameSafe(self, text):
+        keepcharacters = (' ','.','_')
```
Can someone abuse this naming scheme ? Are there any other characters that we might be missing ?

JTSIV1 committed:
Code linting and added user information about folder naming

JTSIV1 committed:
Removed space from accepted name characters

Space was removed to sanitize input and ensure that all inputted names are safe.

alexandermichels reviewed (approved):
Everything hear looks good to me. @taylorziegler Could you look this over and test it? If it works, feel free to merge. 

To install on CyberGISX, run `!pip install --force-reinstall --no-deps git+https://github.com/cybergis/cybergis-compute-python-sdk.git@refs/pull/62/merge` in a cell and then restart the kernel after it installs.

taylorziegler commented:
Looks good to me!
